### PR TITLE
Unquote crate names

### PR DESCRIPTION
**Pull request by richo**
I believe this needs a snapshot to land.

### CODE REVIEW

**Comment by richo** [comment]
Just noticed #23738 right on time :)

File: src/test/run-pass/xcrate-trait-lifetime-param.rs [hunk 1]
```diff
@@ -12,7 +12,7 @@
 
 // pretty-expanded FIXME #23616
 
-extern crate "xcrate-trait-lifetime-param" as other;
+extern crate xcrate-trait-lifetime-param as other;
```
Comment (petrochenkov) [inline src/test/run-pass/xcrate-trait-lifetime-param.rs]: There are hyphens in unquoted crate names here and some other places.

**Comment by alexcrichton** [comment]
@bors: r-

I think @petrochenkov is right in that this won't land as-is. Can you also be sure to run all the tests as well?

**Comment by richo** [comment]
Sure, sorry I PR'd this and then went to bed. I wasn't expecting it to r+ until that snapshot landed. I'll poke today.

**Comment by alexcrichton** [comment]
I've also got a PR where I'm removing parser support, so I can try to just roll this into that as well.

**Comment by richo** [comment]
Reran the script to only unquote non-hyphenated crates. r? @alexcrichton 

(I think this still needs snaps to land, didn't see if they did)

**Comment by alexcrichton** [comment]
Thanks! I've rolled this into #23786 where I took care of the rest of the tests and also handled the compiler support.

**Comment by richo** [comment]
\o/ thanks

**Pull request closed without merging**